js/router.js

```diff
@@ -1,15 +1,153 @@
+      ['transactions', 'transactions'],
+      ['transactions/:tab', 'transactions'],
+      ['*path', 'pageNotFound'],
     ];
 
-    routes.forEach((route) => this.route.apply(this, route));
+    routes.slice(0)
+      .reverse()
+      .forEach((route) => this.route.apply(this, route));
+
+    this.setAddressBarText();
+
+    $(window).on('hashchange', () => {
+      this.setAddressBarText();
+    });
+  }
+
+  setAddressBarText() {
+    if (location.hash.startsWith('#transactions')) {
+      this.pageNavVw.setAddressBar();
+    } else {
+      this.pageNavVw.setAddressBar(location.hash.slice(1));
+    }
+  }
+
+  execute(callback, args) {
+    getPageContainer().empty();
+
+    if (callback) callback.apply(this, args);
+  }
+
+  loadPage(vw) {
+    if (this.currentPage) {
+      this.currentPage.remove();
+    }
+
+    this.currentPage = vw;
+    getPageContainer().append(vw.el);
+  }
+
+  // Temporary fudge, since we're actually hitting the one name api,
+  // but I'd like the guids to match the 2.0 format of being prefaced with Qm.
+  // Once we're getting the data from the server, this method will go bye-bye
+  // and will directly call getGuid from the util module.
+  getGuid(guid) {
+    const deferred = $.Deferred();
+
+    getGuid(guid).done((id) => deferred.resolve(`Qm${id}`))
+      .fail((...args) => deferred.reject(...args));
+
+    return deferred.promise();
+  }
+
+  getUser(guid) {
+    const deferred = $.Deferred();
+    const user = this.usersCl.get(guid);
+
+    // setting timeout to simulate the latency of an async call
+    setTimeout(() => {
+      if (user) {
+        deferred.resolve(user);
+      } else {
+        deferred.reject();
+      }
+    }, parseInt(Math.random() * 1000, 10));
+
+    return deferred.promise();
+  }
+
+  userViaHandle(handle, ...args) {
+    this.getGuid(handle).done((guid) => {
+      this.user(guid, ...args);
+    }).fail(() => {
+      this.userNotFound();
+    });
+  }
+
+  user(guid, tab, ...args) {
+    tab = tab || 'store'; // eslint-disable-line no-param-reassign
+    const pageOpts = { tab };
+
+    if (tab === 'channel') {
+      pageOpts.category = args[0];
+      pageOpts.layer = args[1];
+    }
+
+    this.getUser(guid).done((user) => {
+      const displayArgs = args.filter((arg) => arg !== null).join('/');
+      const handle = user.get('handle');
+
+      this.navigate(`${handle ? `@${handle}` : user.id}/${tab}` +
+        `${displayArgs ? `/${displayArgs}` : ''}`);
+
+      this.loadPage(
+        new UserPage({
+          ...pageOpts,
+          model: user,
+        }).render()
+      );
+    }).fail(() => {
+      this.userNotFound();
+    });
+  }
+
+  transactions(tab) {
+    tab = tab || 'inbound'; // eslint-disable-line no-param-reassign
+
+    this.loadPage(
+      new TransactionsPage({ tab }).render()
+    );
+  }
+
+  userNotFound() {
+    this.loadPage(
+      new TemplateOnly({ template: 'error-pages/userNotFound.html' }).render()
```

Comment:
The template to use is an option of the View, not context that is to be passed to render. The context is actually what data you want to provide to the template. (and the template has no need to know it's own path :)